### PR TITLE
:book:[Docs] How-to: Channel-Based Automatic Updates

### LINKED ISSUE #1129: Docs: Accept automatic upgrades within a specific channel
Write user documentation describing how to keep automatic upgrades to a specific channel. If documentation already exists, review it and ensure it is still accurate and up-to-date.

A/C:
 - Describe what a channel is
 - Instructions to specify a channel for auto-upgrades
 - Call out that you can also use version ranges to further restrict the set (of allowable versions)
 - Call out auto-upgrades from multiple channels

Reach out if you have any questions please reach out on [Slack](https://kubernetes.slack.com/archives/C0181L6JYQ2)

Current documentation is posted up at https://operator-framework.github.io/operator-controller/
New docs should be placed in [docs/drafts](https://github.com/operator-framework/operator-controller/tree/main/docs/drafts)

### PR DESCRIPTION
**Pull request by perdasilva**
# Description

Closes #1129

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: docs/drafts/how-to-channel-based-upgrades.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+Example:
+
+```yaml
+apiVersion: olm.operatorframework.io/v1alpha1
+kind: ClusterExtension
+metadata:
+  name: argocd
+spec:
+  installNamespace: argocd
```
Comment (everettraven) [inline docs/drafts/how-to-channel-based-upgrades.md:13]: There was a recent [PR](https://github.com/operator-framework/operator-controller/pull/1200) that put the `installNamespace` and `serviceAccount` fields under an `install` root field.

**Review by OchiengEd — CHANGES_REQUESTED** [review]
Looks good. I recommend slight rewording

File: docs/drafts/how-to-channel-based-upgrades.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    sourceType: Catalog
+    catalog:
+      packageName: argocd-operator
+      # Automatically upgrade to the latest version found in the preview and dev-preview channels
+      channel: [dev-preview, preview]
+  serviceAccount:
+    name: argocd-installer
+```
+
+Note that the `version` version range can also be set to further restrict the set of possible versions to upgrade to.
```
[the remark this answers is not in the document]
Comment (perdasilva) [reply docs/drafts/how-to-channel-based-upgrades.md]: Nice! Thank you!!

I've updated it and tried to remove the preposition at the end? wdyt?

File: docs/drafts/how-to-channel-based-upgrades.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+kind: ClusterExtension
+metadata:
+  name: argocd
+spec:
+  source:
+    sourceType: Catalog
+    catalog:
+      packageName: argocd-operator
+      # Automatically upgrade to the latest version found in the preview and dev-preview channels
+      channel: [dev-preview, preview]
```
Comment (everettraven) [inline docs/drafts/how-to-channel-based-upgrades.md:18]:
```suggestion
      channels: [dev-preview, preview]
```

File: docs/drafts/how-to-channel-based-upgrades.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+kind: ClusterExtension
+metadata:
+  name: argocd
+spec:
+  source:
+    sourceType: Catalog
+    catalog:
+        packageName: argocd-operator
+  
+      channel: [stable] # Automatically upgrade to the latest version found in ‘stable’
```
Comment (everettraven) [inline docs/drafts/how-to-channel-based-upgrades.md:40]:
```suggestion
      channels: [stable] # Automatically upgrade to the latest version found in ‘stable’
```

Comment (perdasilva) [reply docs/drafts/how-to-channel-based-upgrades.md]: Thank you! Fixed ^^

File: docs/drafts/how-to-channel-based-upgrades.md (hunk 3 above)
Comment (perdasilva) [reply docs/drafts/how-to-channel-based-upgrades.md]: Thank you! Fixed ^^

**Review by everettraven — APPROVED** [review]

**Review by joelanford — APPROVED** [review]

**Pull request merged**